src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -2309,4 +2374,74 @@ private static int log2(int value)
         }
         #endregion
     }
+
+    /// <summary>
+    /// A class that generates an IEqualityComparer for this SortedSet. Requires that the definition of
+    /// equality defined by the IComparer for this SortedSet be consistent with the default IEqualityComparer
+    /// for the type T. If not, such an IEqualityComparer should be provided through the constructor.
+    /// </summary>    
+    internal class SortedSetEqualityComparer<T> : IEqualityComparer<SortedSet<T>> 
+    {
+        private IComparer<T> comparer;
+        private IEqualityComparer<T> e_comparer;
+
+        public SortedSetEqualityComparer() : this(null, null) { }
+
+        public SortedSetEqualityComparer(IComparer<T> comparer) : this(comparer, null) { }
+
+        public SortedSetEqualityComparer(IEqualityComparer<T> memberEqualityComparer) : this(null, memberEqualityComparer) { }
+
+        /// <summary>
+        /// Create a new SetEqualityComparer, given a comparer for member order and another for member equality (these
+        /// must be consistent in their definition of equality)
+        /// </summary>        
+        public SortedSetEqualityComparer(IComparer<T> comparer, IEqualityComparer<T> memberEqualityComparer)
+        {
+            if (comparer == null)
+                this.comparer = Comparer<T>.Default;
+            else
+                this.comparer = comparer;
+            if (memberEqualityComparer == null)
+                e_comparer = EqualityComparer<T>.Default;
+            else
+                e_comparer = memberEqualityComparer;
```

Comment:
Nit: This could be simplified, e.g. (assuming the field renames per my other comment above):

```csharp
public SortedSetEqualityComparer(IComparer<T> comparer, IEqualityComparer<T> memberEqualityComparer)
{
    _comparer = comparer ?? Comparer<T>.Default;
    _memberEqualityComparer = memberEqualityComparer ?? EqualityComparer<T>.Default;
}
```